### PR TITLE
MAINT: Include the function name in all argument error messages

### PR DESCRIPTION
**Pull request by eric-wieser**
Performed using the following regex, replacing with `$0:$1`. Some manual cleanup
was then required

```regex
^[a-z]+_(\w+)(?:(?!\}).*\n)+.*PyArg_ParseTuple(AndKeywords)?\(args[^"]+"[^:"]+(?=")
```

### CODE REVIEW

File: numpy/core/src/multiarray/multiarraymodule.c [hunk 1]
```diff
@@ -3263,7 +3263,7 @@ array_can_cast_safely(PyObject *NPY_UNUSED(self), PyObject *args,
     NPY_CASTING casting = NPY_SAFE_CASTING;
     static char *kwlist[] = {"from", "to", "casting", NULL};
 
-    if(!PyArg_ParseTupleAndKeywords(args, kwds, "OO&|O&", kwlist,
+    if(!PyArg_ParseTupleAndKeywords(args, kwds, "OO&|O&:can_cast", kwlist,
```
Comment (eric-wieser) [inline numpy/core/src/multiarray/multiarraymodule.c]: It's a little annoying how sometimes the C and python function names are inconsistent, like here (`can_cast` vs `can_cast_safely`)

File: numpy/core/src/multiarray/multiarraymodule.c [hunk 2]
```diff
@@ -2855,7 +2855,7 @@ array_fastCopyAndTranspose(PyObject *NPY_UNUSED(dummy), PyObject *args)
 {
     PyObject *a0;
 
-    if (!PyArg_ParseTuple(args, "O", &a0)) {
+    if (!PyArg_ParseTuple(args, "O:fastCopyAndTranspose", &a0)) {
```
Comment (juliantaylor) [inline numpy/core/src/multiarray/multiarraymodule.c]: this is `_fastCopyAndTranspose`

**Comment by juliantaylor** [comment]
looks good modulo one nitpick

while greatly appreciated you should slow down a little with the changes to avoid a too fast burnout, also we don't have enough review time to keep up.

File: numpy/core/src/multiarray/multiarraymodule.c (hunk 2 above)
Comment (eric-wieser) [inline numpy/core/src/multiarray/multiarraymodule.c]: Good catch

**Comment by charris** [comment]
> you should slow down a little with the changes

My motto, "Gather ye rosebuds while ye may." Few last more than a few years in open source, the needs of life eventually take precedence...

**Pull request merged**